### ISSUE TITLE
is it possible to produce message synchronously?

### ISSUE THREAD

**Issue by KarthikNedunchezhiyan** [fede1024/rust-rdkafka#286]
Hi team, My application is based on event loop using mio poll. I am not using tokio runtime. It is possible to use this crate synchronously to produce messages?

**Comment by benesch** [comment]
There is a good bit of exposition on the subject in the producer documentation: https://docs.rs/rdkafka/0.24.0/rdkafka/producer/index.html

The tl;dr is that librdkafka is itself asynchronous, and so there is not a convenient API for synchronous production. That said, you don't need to use Tokio if you don't want to. Neither the [`BaseProducer`](https://docs.rs/rdkafka/0.24.0/rdkafka/producer/base_producer/struct.BaseProducer.html) or the [`ThreadedProducer`](https://docs.rs/rdkafka/0.24.0/rdkafka/producer/base_producer/struct.ThreadedProducer.html) require Tokio (or any other asynchronous runtime). You'll just need to do some plumbing to wait for "delivery report" notifications via the [`ProducerContext::delivery`](https://docs.rs/rdkafka/0.24.0/rdkafka/producer/base_producer/trait.ProducerContext.html#tymethod.delivery) callback.
